### PR TITLE
[MM-55329] Convert ./plugins/test/post_type.test.jsx from Class Component to Function Component

### LINKED ISSUE #25358: Convert ./plugins/test/post_type.test.jsx from Class Component to Function Component
We want to move most of our codebase from Class Components to Function Components. This decision is based on [React recommendations](https://react.dev/reference/react/Component).

The task is to update the component to be a functional component, without any difference in functionality. Also, all related tests that fail may need refactoring.

Here are a few examples of the required changes:
https://github.com/mattermost/mattermost/pull/24187
https://github.com/mattermost/mattermost/pull/24210
https://github.com/mattermost/mattermost/pull/24183
https://github.com/mattermost/mattermost/pull/24181

If the tests do not work because they rely on the behavior of Class Components they also have to be migrated to functionaly test the same behavior. We prefer the use of `react-testing-library` for new tests, but it is not required.

When submitting the PR, please make sure to identify where and when the component shows, for QA to properly test the changes.

----

If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA LINK: MM-55329

### PR DESCRIPTION
**Pull request by rohitkbc**
#### Summary
Fixes #25358 

#### Ticket Link
JIRA LINK: [MM-55329](https://mattermost.atlassian.net/browse/MM-55329)

```release-note
NONE
```

[MM-55329]: https://mattermost.atlassian.net/browse/MM-55329?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webapp/channels/src/plugins/test/post_type.test.tsx [hunk 1]
```diff
@@ -8,11 +8,9 @@ import {Preferences} from 'mattermost-redux/constants';
 
 import PostMessageView from 'components/post_view/post_message_view/post_message_view';
 
-class PostTypePlugin extends React.PureComponent {
-    render() {
-        return <span id='pluginId'>{'PostTypePlugin'}</span>;
-    }
-}
+const PostTypePlugin: React.FunctionComponent = () => (
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/plugins/test/post_type.test.tsx:11]:
```suggestion
const PostTypePlugin = () => (
```
nit we can remove that

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Review by devinbinnie — APPROVED** [review]
LGTM other than Zubair's comment. Thanks @rohitkbc!

**Comment by larkox** [comment]
@rohitkbc Are you still working on this? Can you take a look to the review comments?

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]
@rohitkbc can you please the failing CI checks

**Review by M-ZubairAhmed — APPROVED** [review]

**Pull request merged**